RecoTracker/LSTCore/src/alpaka/TrackCandidate.h

```diff
@@ -456,6 +528,216 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
+            continue;
+
+          // Embedding distance
+          float embedDistance2 = 0.f;
+          CMS_UNROLL_LOOP
+          for (unsigned int embedIndex = 0; embedIndex < Params_T5::kEmbed; ++embedIndex) {
+            const float diff = quintuplets.t5Embed()[baseQuintupletIndex][embedIndex] -
+                               quintuplets.t5Embed()[candidateQuintupletIndex][embedIndex];
+            embedDistance2 += diff * diff;
+          }
+          if (embedDistance2 > 1.0f)
+            continue;
+
+          // Hit matching against base T5 hits
+          int sharedHitCount = 0;
+          int unmatchedLayerSlot = -1;
+
+          countSharedT5HitsAndFindUnmatched(
+              quintuplets, candidateQuintupletIndex, baseQuintupletIndex, sharedHitCount, unmatchedLayerSlot);
+
+          if (sharedHitCount < lst::kT5DuplicateMinSharedHits)
+            continue;
+          if (unmatchedLayerSlot < 0)
+            continue;
+
+          // Candidate score is the T5 DNN output
+          const float candidateScore = quintuplets.dnnScore()[candidateQuintupletIndex];
+
+          // New OT logical layer from the candidate
+          const uint8_t newLogicalLayer = quintuplets.logicalLayers()[candidateQuintupletIndex][unmatchedLayerSlot];
+          const int logicalLayerBin = static_cast<int>(newLogicalLayer) - 1;  // 0..kLogicalOtLayers-1
+
+          // Atomic CAS update on best score for this logical layer
+          bool updateDone = false;
+          while (!updateDone) {
+            const int oldScoreBits = sharedBestScoreBits[logicalLayerBin];
+            const float oldScore = bitCastIntToFloat(oldScoreBits);
+
+            if (candidateScore <= oldScore) {
+              updateDone = true;
+            } else {
+              const int newScoreBits = bitCastFloatToInt(candidateScore);
+              const int assumedBits = alpaka::atomicCas(
+                  acc, &sharedBestScoreBits[logicalLayerBin], oldScoreBits, newScoreBits, alpaka::hierarchy::Threads{});
+              if (assumedBits == oldScoreBits) {
+                sharedBestQuintupletIndex[logicalLayerBin] = static_cast<int>(candidateQuintupletIndex);
+                sharedBestLayerSlotInQuintuplet[logicalLayerBin] = unmatchedLayerSlot;
+                updateDone = true;
+              }
+            }
+          }
+        }
+      }
+
+      // One thread per block finalizes by actually extending the TC
+      alpaka::syncBlockThreads(acc);
+
+      if (cms::alpakatools::once_per_block(acc)) {
+        CMS_UNROLL_LOOP
+        for (int logicalLayerBin = 0; logicalLayerBin < lst::kLogicalOtLayers; ++logicalLayerBin) {
+          const int bestQuintupletIndex = sharedBestQuintupletIndex[logicalLayerBin];
+          if (bestQuintupletIndex < 0)
+            continue;
+
+          const int bestQuintupletLayerSlot = sharedBestLayerSlotInQuintuplet[logicalLayerBin];
+          const uint8_t logicalLayer = static_cast<uint8_t>(logicalLayerBin + 1);  // 1..11
```

Comment:
Was this added to avoid some compiler warning?
this short/char 1 is repeated in a few places, perhaps define kOneU8 or smth.